### PR TITLE
Fixes #24565 - Unable to resolve dependencies for gem rest-client

### PR DESCRIPTION
**Pull request by apuntamb**

### CODE REVIEW

**Comment by apuntamb** [comment]
@lzap mind taking  a look at this? the rest-client version conflicts with the one needed for core.

**Comment by lzap** [comment]
Tests are very broken (not by this PR). I need to find some time to take a look on this.

**Comment by apuntamb** [comment]
@lzap Sure. Thanks for looking into it.

**Comment by lzap** [comment]
So I'd prefer to drop all explicit versions, does it work for you? We introduced this because of https://projects.theforeman.org/issues/8907 but this appears to be fixed now.

Here is a diff that does that plus it fixes also broken tests:

```diff
diff --git a/lib/smart_proxy_discovery/discovery_main.rb b/lib/smart_proxy_discovery/discovery_main.rb
index 2fa8688..fca4976 100644
--- a/lib/smart_proxy_discovery/discovery_main.rb
+++ b/lib/smart_proxy_discovery/discovery_main.rb
@@ -8,8 +8,6 @@ module Proxy::Discovery
 
   class << self
     CREATE_DISCOVERED_HOST_PATH = '/api/v2/discovered_hosts/facts'
-    SCHEME = Proxy::Discovery::Plugin.settings.node_scheme
-    PORT = Proxy::Discovery::Plugin.settings.node_port
 
     def create_discovered_host(request)
       foreman_request = Proxy::HttpRequest::ForemanRequest.new()
@@ -51,7 +49,9 @@ module Proxy::Discovery
     private
 
     def generate_url(ip)
-      "#{SCHEME}://#{ip}:#{PORT}"
+      scheme = Proxy::Discovery::Plugin.settings.node_scheme
+      port = Proxy::Discovery::Plugin.settings.node_port
+      "#{scheme}://#{ip}:#{port}"
     end
 
     def get_rest_client(url)
diff --git a/smart_proxy_discovery.gemspec b/smart_proxy_discovery.gemspec
index 67291c7..9c1d8c3 100644
--- a/smart_proxy_discovery.gemspec
+++ b/smart_proxy_discovery.gemspec
@@ -13,12 +13,12 @@ Gem::Specification.new do |s|
   s.homepage = 'http://github.com/theforeman/smart_proxy_discovery'
   s.license = 'GPLv3'
 
+  s.add_development_dependency('test-unit')
   s.add_development_dependency('rake')
-  # http://projects.theforeman.org/issues/9061
-  s.add_development_dependency('rack', '< 1.6')
+  s.add_development_dependency('rack')
   s.add_development_dependency('rack-test')
   s.add_development_dependency('mocha')
   s.add_development_dependency('webmock')
 
-  s.add_runtime_dependency('rest-client', ['> 1.6.2', '< 2'])
+  s.add_runtime_dependency('rest-client')
 end
```

**Comment by apuntamb** [comment]
@lzap Thanks for the quick solution. It works for me.

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 gemspec
 
-gem 'rest-client', '< 1.7' if RUBY_VERSION == '1.8.7' # 1.7.0 drops support for ruby 1.8.7
+gem 'rest-client', '>= 2.0.0', '< 3' if RUBY_VERSION > '1.8.7'
```
Comment (lzap) [inline Gemfile]: Can you just `gem 'rest-client'` here?

Comment (apuntamb) [reply Gemfile]: Sure. done!

**Review by lzap — APPROVED** [review]
And it's green again thanks!

**Pull request merged**